cloudinit/distros/networking.py

```diff
@@ -68,7 +71,7 @@ def get_interfaces(self) -> list:
         return net.get_interfaces()
 
     def get_interfaces_by_mac(self) -> dict:
-        return net.get_interfaces_by_mac()
+        return net.get_interfaces_by_mac(self.blacklist_drivers)
```

Comment:
I suggest passing this as a kwarg, further changes to the args won't need to worry about positional parameter placement.

```suggestion
        return net.get_interfaces_by_mac(blacklist_drivers=self.blacklist_drivers)
```